### ISSUE TITLE
Search page shows latest chronological version, not latest semver

### ISSUE THREAD

**Issue by repnop** [rust-lang/docs.rs#708]
As of the creation of this issue (2020-04-10) the [search page](https://docs.rs/releases/search?query=string-interner) for `string-interner` looks like this:

![image](https://user-images.githubusercontent.com/24203105/79012511-746c6c80-7b34-11ea-99cf-9fc666408cb5.png)

and directs you to the page for version `0.6.4`. However, the latest version of `string-interner` is `0.7.1`, which is detected and displays the warning banner at the top to redirect you to `0.7.1`.

Perhaps this is caused by the... very awkward release dates? Somehow the latest release from a semver point of view is `0.7.1` but date-wise its actually `0.6.4`

From crates.io:
```
Versions

    0.7.1 Sep 1, 2019
    0.7.0 Aug 7, 2018 yanked
    0.6.4 Sep 3, 2019
    0.6.3 Sep 19, 2017 yanked
    0.6.2 Aug 13, 2017 yanked
```

**Comment by jyn514** [comment]
> Perhaps this is caused by the... very awkward release dates? Somehow the latest release from a semver point of view is 0.7.1 but date-wise its actually 0.6.4

Yes, that sounds like the cause. I don't see a great way to fix this since currently searches are done [using SQL only](https://github.com/rust-lang/docs.rs/blob/13eb4e80e16045fa11dcf8292150dea637d17a01/src/web/releases.rs#L247). We could normalize the search results after already picking a crate to show, but I'm a little hesitant to do that without a long-term plan for improving search results.

**Comment by jyn514** [comment]
I think this may have been fixed by https://github.com/rust-lang/docs.rs/pull/1521, but I don't have an easy way to check.

**Comment by syphar** [comment]
> I think this may have been fixed by #1521, but I don't have an easy way to check.

not really, in #1521 we still order by release-time for the version. #1546 will fix it correctly IMHO

**Comment by syphar** [comment]
Fixed with #1546 and  #1567
